advanced_server.rb

```diff
@@ -0,0 +1,314 @@
+          name: 'Octo Rubocop',
+          status: 'in_progress',
+          started_at: Time.now.utc.iso8601
+      })
+
+      # ***** RUN A CI TEST *****
+      # This is where we would kick off our CI process. Ideally this would be performed async, so we could
+      # return immediately. But for now we'll do a simulated CI process syncronously, and update the check run right here..
+
+      full_repo_name = @payload['repository']['full_name']
+      repository     = @payload['repository']['name']
+      head_sha       = @payload['check_run']['head_sha']
+      repository_url = @payload['repository']['html_url']
+
+      clone_repository(full_repo_name, repository, head_sha)
+
+      @report = `rubocop '#{repository}/*' --format json` # --auto-correct`
+      `rm -rf #{repository}`
+      @output = JSON.parse @report
+      annotations = []
+
+      if @output["summary"]["offense_count"] == 0
+        conclusion = 'success'
+      else
+        conclusion = 'neutral'
+        @output["files"].each do |file|
+          file_path = file["path"].gsub(/#{repository}\//,'')
+          blob_href = "#{repository_url}/blob/#{head_sha}/#{file_path}"
+          warning_level = 'notice'
+          file["offenses"].each do |offense|
+            start_line   = offense["location"]["start_line"]
+            end_line     = offense["location"]["last_line"]
+            start_column = offense["location"]["start_column"]
+            end_column   = offense["location"]["last_column"]
+            message      = offense["message"]
+            annotation = {
+              "filename" => file_path,
+              "blob_href" => blob_href,
+              "start_line" => start_line,
+              "end_line" => end_line,
+              "start_column" => start_column,
+              "end_column" => end_column,
+              "warning_level" => warning_level,
+              "message" => message
+            }
+            annotations.push(annotation)
+          end
+        end
+      end
+
+      summary = "Octoc Rubocop summary\n-Offense count: #{@output["summary"]["offense_count"]}\n-File count: #{@output["summary"]["target_file_count"]}\n-Target file count: #{@output["summary"]["inspected_file_count"]}}"
+      details = "Octoc Rubocop version: #{@output["metadata"]["rubocop_version"]}"
+
+      # Now, mark the check run as complete! And if there are warnings, share them
+      updated_check_run = @installation_client.patch("repos/#{@payload['repository']['full_name']}/check-runs/#{@payload['check_run']['id']}", {
+          accept: 'application/vnd.github.antiope-preview+json', # This header is necessary for beta access to Checks API
+          name: 'Octo Rubocop',
+          status: 'completed',
+          conclusion: conclusion,
+          completed_at: Time.now.utc.iso8601,
+          output: {
+            title: "Octoc Rubocop",
+            summary: summary,
+            details: details,
```

Comment:
According to [the docs](https://developer.github.com/v3/checks/runs/#parameters-1), `details:` should probably be `text:`.

Reply:
Oops! Yes, that is not the correct param name. Updated!